### PR TITLE
Add implementations of NetworkBehaviour for ping

### PR DESCRIPTION
**Pull request by tomaka**
cc #596 

Trivial layers that just apply a protocol handler to each connection.

### CODE REVIEW

File: protocols/ping/src/dial_layer.rs [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+}
+
+impl<TSubstream> Default for PeriodicPingBehaviour<TSubstream> {
+    #[inline]
+    fn default() -> Self {
+        PeriodicPingBehaviour::new()
+    }
+}
+
+impl<TSubstream> NetworkBehavior for PeriodicPingBehaviour<TSubstream>
```
Comment (twittner) [inline protocols/ping/src/dial_layer.rs]: Just noticed that our spelling of behaviour is inconsistent.

Comment (tomaka) [reply protocols/ping/src/dial_layer.rs]: Yeah, I noticed that and want to fix the typo once everything is merged.

File: protocols/ping/src/dial_layer.rs [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+
+use futures::prelude::*;
+use libp2p_core::nodes::{ConnectedPoint, NetworkBehavior, NetworkBehaviorAction};
+use libp2p_core::{nodes::protocols_handler::ProtocolsHandler, PeerId};
+use std::marker::PhantomData;
+use tokio_io::{AsyncRead, AsyncWrite};
+use void::Void;
+use PeriodicPingHandler;
+
+/// Network behaviour that handles receiving to pings sent by other nodes.
```
Comment (dvdplm) [inline protocols/ping/src/dial_layer.rs]: spurious "to"

File: protocols/ping/src/listen_layer.rs [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+
+use futures::prelude::*;
+use libp2p_core::nodes::{ConnectedPoint, NetworkBehavior, NetworkBehaviorAction};
+use libp2p_core::{nodes::protocols_handler::ProtocolsHandler, PeerId};
+use std::marker::PhantomData;
+use tokio_io::{AsyncRead, AsyncWrite};
+use void::Void;
+use PingListenHandler;
+
+/// Network behaviour that handles receiving to pings sent by other nodes.
```
Comment (dvdplm) [inline protocols/ping/src/listen_layer.rs]: spurious "to"

**Pull request merged**